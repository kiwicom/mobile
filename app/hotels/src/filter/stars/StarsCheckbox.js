// @flow

import * as React from 'react';
import { View, Text, StyleSheet, TouchableWithoutFeedback } from 'react-native';
import { Color, Stars, Icon } from '@kiwicom/react-native-app-common';

type Props = {|
  onPress: () => void,
  stars?: number,
  isChecked?: boolean,
  text?: string,
  // $FlowFixMeProps
  style?: Object,
|};

export default function StarsCheckbox(props: Props) {
  return (
    <TouchableWithoutFeedback onPress={props.onPress}>
      <View style={[styles.wrapper, props.style]}>
        <View style={styles.label}>
          {props.stars ? (
            <Text style={styles.stars}>
              <Stars rating={props.stars} />
            </Text>
          ) : (
            <Text style={styles.text}>{props.text}</Text>
          )}
        </View>
        {props.isChecked && (
          <View style={styles.check}>
            <Icon name="check" size={26} color={Color.brand} />
          </View>
        )}
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    paddingTop: 8,
    paddingBottom: 8,
  },
  label: {
    flex: 3,
    justifyContent: 'center',
    height: 26,
  },
  stars: {
    fontSize: 16,
    color: Color.brand,
  },
  hotels: {
    marginLeft: 3,
    fontSize: 12,
    color: Color.grey.$600,
  },
  text: {
    marginLeft: 3,
    fontSize: 16,
    fontWeight: '300',
  },
  check: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
});
