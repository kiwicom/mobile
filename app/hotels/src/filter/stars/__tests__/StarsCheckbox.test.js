// @flow

import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import StarsCheckbox from '../StarsCheckbox';

const renderer = new ShallowRenderer();

describe('StarsCheckbox', () => {
  it('render unchecked with 1 star and 1 hotel', () => {
    const starsCheckbox = (
      <StarsCheckbox stars={1} hotels={1} onPress={jest.fn()} />
    );
    expect(renderer.render(starsCheckbox)).toMatchSnapshot();
  });

  it('render unchecked with stars and hotels', () => {
    const starsCheckbox = (
      <StarsCheckbox stars={3} hotels={10} onPress={jest.fn()} />
    );
    expect(renderer.render(starsCheckbox)).toMatchSnapshot();
  });

  it('render checked', () => {
    const starsCheckbox = (
      <StarsCheckbox
        stars={5}
        hotels={123}
        isChecked={true}
        onPress={jest.fn()}
      />
    );
    expect(renderer.render(starsCheckbox)).toMatchSnapshot();
  });

  it('render text instead of stars', () => {
    const starsCheckbox = (
      <StarsCheckbox text="Some text" hotels={123} onPress={jest.fn()} />
    );
    expect(renderer.render(starsCheckbox)).toMatchSnapshot();
  });
});
