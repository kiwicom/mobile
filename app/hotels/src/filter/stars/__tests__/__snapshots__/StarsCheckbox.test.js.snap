// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarsCheckbox render checked 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#0097a9",
            "fontSize": 16,
          }
        }
      >
        <Stars
          rating={5}
        />
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Icon
        color="#0097a9"
        name="check"
        size={26}
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`StarsCheckbox render text instead of stars 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "300",
            "marginLeft": 3,
          }
        }
      >
        Some text
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`StarsCheckbox render unchecked with 1 star 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#0097a9",
            "fontSize": 16,
          }
        }
      >
        <Stars
          rating={1}
        />
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`StarsCheckbox render unchecked with 3 stars 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "height": 26,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#0097a9",
            "fontSize": 16,
          }
        }
      >
        <Stars
          rating={3}
        />
      </Text>
    </View>
  </View>
</TouchableWithoutFeedback>
`;
